magda-web-client/src/actions/datasetSearchActions.js

```diff
@@ -10,10 +10,14 @@ import type {
     SearchAction
 } from "../helpers/datasetSearch";
 
-export function requestResults(apiQuery: string): SearchAction {
+export function requestResults(
+    apiQuery: string,
+    json: DataSearchJson
```

Comment:
(comment 2/3, read the next one first!)

... so now you can change this to

```javascript
export function requestResults(
    query: Object
): SearchAction {
```

Accepting the query object, then pass that on below instead of `{apiQuery, json}`